AVMFritzbox/AVMFritzbox.php

```diff
@@ -1,5 +1,209 @@
+            $this->url("statusInfo"),
+            $this->getAttrs("statusInfo")
+        );
+		echo $test->status;
+	}
+
+	public function livestats()
+	{
+		$status = "inactive";
+
+        $statusInfo = parent::execute(
+			$this->url("statusInfo"),
+			$this->getAttrs("statusInfo")
+		);
+		$statusInfoDetails = simplexml_load_string($statusInfo->getBody())->children('s', true)->children('u', true)->children();
+
+        $linkProperties = parent::execute(
+			$this->url("linkProperties"),
+			$this->getAttrs("linkProperties")
+		);
+		$linkPropertiesDetails = simplexml_load_string($linkProperties->getBody())->children('s', true)->children('u', true)->children();
+
+        $addonInfo = parent::execute(
+			$this->url("addonInfo"),
+			$this->getAttrs("addonInfo")
+		);
+		$addonInfoDetails = simplexml_load_string($addonInfo->getBody())->children('s', true)->children('u', true)->children();
+
+		$data = ["visiblestats" => []];
+
+		if ($statusInfoDetails && $linkPropertiesDetails && $addonInfoDetails) {
+			foreach ($this->config->availablestats as $stat) {
+                if (!isset(self::getAvailableStats()[$stat])) {
+					continue;
+				}
+
+                $newstat = new \stdClass();
+				$newstat->title = self::getAvailableStats()[$stat];
+
+                switch ($stat) {
+                    case "NewConnectionStatus":
+                    case "NewUptime":
+                        $newstat->value = self::formatValueUsingStat(
+                            $stat,
+                            $statusInfoDetails->$stat
+                        );
+                        break;
+                    case "NewLayer1DownstreamMaxBitRate":
+                    case "NewLayer1UpstreamMaxBitRate":
+                        $newstat->value = self::formatValueUsingStat(
+                            $stat,
+                            $linkPropertiesDetails->$stat
+                        );
+                        break;
+                    case "NewByteReceiveRate":
+                    case "NewByteSendRate":
+                    case "NewX_AVM_DE_TotalBytesReceived64":
+                    case "NewX_AVM_DE_TotalBytesSent64":
+                        $newstat->value = self::formatValueUsingStat(
+                            $stat,
+                            $addonInfoDetails->$stat
+                        );
+                        break;
+                    default:
+                        $newstat->value = self::formatValueUsingStat(
+                            $stat,
+                            $addonInfoDetails->$stat
+                        );
+                }
+
+				$data["visiblestats"][] = $newstat;
+			}
+		}
+
+		return parent::getLiveStats($status, $data);
+	}
+
+	public function url($endpoint)
+	{
+        if ($endpoint == "statusInfo") {
+            $endpoint = "WANIPConn1";
+        } else {
+            $endpoint = "WANCommonIFC1";
+        }
+
+		$api_url = "http://fritz.box:49000/igdupnp/control/{$endpoint}";
```

Comment:
Fritzbox is always resolving this? I assume lot of Heimdall users also use PiHole that won't resolve it. Maybe add an option to provide here the address from the UI.

Reply:
If this is the case they can still use a local DNS entry inside PiHole. I have also a PiHole instance and no problems with it (without the local DNS entry).